scan-command/src/main/java/io/ballerina/scan/internal/ScanCmd.java

```diff
@@ -244,6 +314,19 @@ private Optional<Project> getProject() {
         }
     }
 
+    private URLClassLoader loadPlatformPlugins(List<String> jarPaths) {
+        List<URL> jarUrls = new ArrayList<>();
+        jarPaths.forEach(jarPath -> {
+            try {
+                URL jarUrl = Path.of(jarPath).toUri().toURL();
+                jarUrls.add(jarUrl);
+            } catch (MalformedURLException ex) {
+                throw new ScanToolException(ex.getMessage());
+            }
+        });
+        return new URLClassLoader(jarUrls.toArray(new URL[0]), this.getClass().getClassLoader());
```

Comment:
Avoid creating temporary zero length arrays
```suggestion
        return new URLClassLoader(jarUrls.toArray(URL[]::new), this.getClass().getClassLoader());
```